From 22c7237238d597562a6f77a90a43e0cd6ee8abdd Mon Sep 17 00:00:00 2001
From: Pranav <npv12@iitbbs.ac.in>
Date: Sun, 31 Jul 2022 23:24:12 +0530
Subject: [PATCH] drm/msm: Fix issues with `__crc_gsi_write_channel_scratch`

23:18:29 WARNING: EXPORT symbol "gsi_write_channel_scratch" [vmlinux] version generation failed, symbol will not be versioned.
23:18:29 ld.lld: error: relocation R_AARCH64_ABS32 cannot be used against symbol '__crc_gsi_write_channel_scratch'; recompile with -fPIC
23:18:30 >>> defined in built-in.o(drivers/platform/msm/gsi/gsi.o)
23:18:30 >>> referenced by gsi.c
23:18:30 >>>               drivers/platform/msm/gsi/gsi.o:(___kcrctab+gsi_write_channel_scratch+0x0) in archive built-in.o
23:18:30 make[1]: *** [/home2/npv12/kernel/strix/Makefile:1175: vmlinux] Error 1
23:18:30 make[1]: Leaving directory '/home2/npv12/kernel/strix/out'
23:18:30 make: *** [Makefile:146: sub-make] Error 2

Signed-off-by: Pranav <npv12@iitbbs.ac.in>
---
 drivers/platform/msm/gsi/gsi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/platform/msm/gsi/gsi.c b/drivers/platform/msm/gsi/gsi.c
index c62aebc8ef0a..e468bfb76089 100644
--- a/drivers/platform/msm/gsi/gsi.c
+++ b/drivers/platform/msm/gsi/gsi.c
@@ -2623,7 +2623,7 @@ static void __gsi_read_wdi3_channel_scratch2_reg(unsigned long chan_hdl,
 }
 
 
-static union __packed gsi_channel_scratch __gsi_update_mhi_channel_scratch(
+static union gsi_channel_scratch __gsi_update_mhi_channel_scratch(
 	unsigned long chan_hdl, struct __packed gsi_mhi_channel_scratch mscr)
 {
 	union __packed gsi_channel_scratch scr;
